lib/faraday/multipart/middleware.rb

```diff
@@ -97,6 +127,17 @@ def process_params(params, prefix = nil, pieces = nil, &block)
           end
         end
       end
+
+      # Determines and provides the multipart mime type for the request.
+      #
+      # @return [String] the multipart mime type
+      def mime_type
+        @mime_type ||= if @options[:content_type].to_s.start_with?(/multipart\/.+/)
+          @options[:content_type].to_s
+        else
+          'multipart/form-data'
+        end
+      end
```

Comment:
I figured as long as the Content-Type started with multipart that is mainly what we cared about and it keeps this middleware flexible.